vertical-pod-autoscaler/pkg/utils/vpa/api.go

```diff
@@ -49,32 +50,34 @@ func patchVpa(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string,
 	return vpaClient.Patch(vpaName, types.JSONPatchType, bytes)
 }
 
-// InitVpaStatus inserts into VPA CRD object an empty VerticalPodAutoscalerStatus object, so later it can be filled with data.
-func InitVpaStatus(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpaName string) (result *vpa_types.VerticalPodAutoscaler, err error) {
-	return patchVpa(vpaClient, vpaName, []patchRecord{{
-		Op:    "add",
-		Path:  "/status",
-		Value: vpa_types.VerticalPodAutoscalerStatus{},
-	},
-	})
-}
+// UpdateVpaStatus updates the status field of the VPA API object.
+// It prevents race conditions by verifying that the lastUpdateTime of the
+// API object and its model representation are equal.
+func UpdateVpaStatus(vpaClient vpa_api.VerticalPodAutoscalerInterface, vpa *model.Vpa) (result *vpa_types.VerticalPodAutoscaler, err error) {
+	status := vpa_types.VerticalPodAutoscalerStatus{
+		LastUpdateTime: metav1.Now(),
```

Comment:
To be perfectly sure this solves races, you need to check that each patch strictly increases LastUpdateTime. While this should be true in sane worlds (but not e.g. if different nodes in the cluster have desynchronized clocks), I would add the if statement .

Reply:
Done.